src/Microsoft.Extensions.Options/OptionsSnapshot.cs

```diff
@@ -21,10 +23,18 @@ public class OptionsSnapshot<TOptions> : IOptionsSnapshot<TOptions> where TOptio
         /// </summary>
         /// <param name="cache">The cache to use.</param>
         /// <param name="factory">The factory to use to create options.</param>
-        public OptionsSnapshot(IOptionsCache<TOptions> cache, IOptionsFactory<TOptions> factory)
+        /// <param name="changeSources">The change token sources used to detect options changes.</param>
+        public OptionsSnapshot(IOptionsCache<TOptions> cache, IOptionsFactory<TOptions> factory, IEnumerable<IOptionsChangeTokenSource<TOptions>> changeSources)
         {
             _cache = cache;
             _factory = factory;
+
+            foreach (var source in changeSources)
```

Comment:
This is a little bit weird, basically we are relying on someone using IOptionsSnapshot to 'turn on' change tracking.  But I think this is ok, because if no one is listening for changes, then we shouldn't turn on the cache invalidation...